pandas/io/json/_normalize.py

```diff
@@ -228,13 +228,17 @@ def json_normalize(
     Returns normalized data with columns prefixed with the given string.
     """
 
-    def _pull_field(js, spec):
-        result = js
+    def _pull_field(js: Dict[str, Any], spec: Union[List, str]) -> List:
         if isinstance(spec, list):
+            _res = js
             for field in spec:
-                result = result[field]
+                _res = _res[field]
+            result = cast(List, _res)
```

Comment:
this is very odd to do a cast here. instead of this why don't you just do

```
result: List
```
at the top

Reply:
cc @simonjayhawkins

Reply:
@jreback the cast is here to satisfy mypi. It is unable to determine the right type otherwise (it will assume a dict) the cast is a no-op at runtime

Reply:
> this is very odd to do a cast here. instead of this why don't you just do
> 
> ```
> result: List
> ```
> 
> at the top

I _think_ that it should be `result: Any` at the top. the isinstance check will allow mypy to narrow the types and ensure the correct return type.

Reply:
Given that `str` is acceptable as a return from `js[field]` I will change it to `result: Any` If the tests pass.